sa/sa.go

```diff
@@ -1303,19 +1299,63 @@ func (ssa *SQLStorageAuthority) NewOrder(ctx context.Context, req *corepb.Order)
 			OrderID:      order.ID,
 			ReversedName: ReverseName(name),
 		}
-		err = tx.Insert(reqdName)
-		if err != nil {
-			err = Rollback(tx, err)
-			return nil, err
+		if err := tx.Insert(reqdName); err != nil {
+			return nil, Rollback(tx, err)
 		}
 	}
 
+	if err := tx.Commit(); err != nil {
+		return nil, err
+	}
+
+	req.Id = &order.ID
+	return req, nil
+}
+
+// SetOrderProcessing updates a provided *corepb.Order in pending status to be
+// in processing status by updating the status field of the corresponding Order
+// table row in the DB. The updated Order is returned on successful update. We
+// avoid introducing a general purpose "Update this order" RPC to ensure we have
+// minimally permissive RPCs.
+func (ssa *SQLStorageAuthority) SetOrderProcessing(ctx context.Context, req *corepb.Order) (*corepb.Order, error) {
+	tx, err := ssa.dbMap.Begin()
+	if err != nil {
+		return nil, err
+	}
+
+	result, err := tx.Exec(`
+		UPDATE orders
+		SET status = ?
+		WHERE id = ?
+		AND status = ?`,
+		string(core.StatusProcessing),
+		*req.Id,
+		string(core.StatusPending))
+	if err != nil {
+		err = berrors.InternalServerError("error updating order to processing status")
+		err = Rollback(tx, err)
+		return nil, err
+	}
+
+	n, err := result.RowsAffected()
+	if err != nil || n == 0 {
+		err = berrors.InternalServerError("no order updated to processing status")
+		err = Rollback(tx, err)
+		return nil, err
+	}
+
 	err = tx.Commit()
 	if err != nil {
 		return nil, err
 	}
 
-	req.Id = &order.ID
+	// Set the status to Processing in the request ob we return since this is the
```

Comment:
s/ob//

Also, do we definitely need to return the order? It seems like just returning an error or nil error would be clearer, since we wouldn't be claiming to fill out the order. I feel like we've run into issues elsewhere where we pass around partially-filled objects (like an authz with just a RegistrationID passed to the VA).

Reply:
> Also, do we definitely need to return the order? It seems like just returning an error or nil error would be clearer, since we wouldn't be claiming to fill out the order.

We don't - happy to rejig the two SA RPCs to not return the Order. Might as well save some bytes on the wire :-)

Reply:
Fixed with https://github.com/letsencrypt/boulder/pull/3169/commits/b74f15974e86fa0326d1fd28a7ff08a65af65cb1